backend/frontend/views.py

```diff
@@ -69,6 +71,28 @@ def is_super(user):
     return user.is_superuser
 
 
+def can_view_user_board(logged_in_user):
+    request = get_current_request()
+    viewed_user_id = request.resolver_match.kwargs.get("user_id")
+
+    if logged_in_user.id == viewed_user_id or logged_in_user.is_superuser:
+        return True
+
+    viewed_user_obj = get_object_or_404(User, pk=viewed_user_id)
+    viewed_user_teams = viewed_user_obj.teams()
+
+    if len(viewed_user_teams):
+        checker = ObjectPermissionChecker(logged_in_user)
+        checker.prefetch_perms(viewed_user_teams)
+        for view_permission in Team.PERMISSION_VIEW:
```

Comment:
This can still be more efficient. Here's something to try out:

```
def work():
     answer = random.choice([True,False])
     print(f"work - {answer}")
     return answer

# what is the difference between the following:
any([work() for _ in range(10)])
any((work() for _ in range(10)))
```

One is a list comprehension, and the other one uses a tuple. The second one is more like a generator. It wont do ALL the work up front.

Now imagine if `work` was an expensive function, eg, what if it queried a database..

TLDR; a tuple is a way better choice here. Not just slightly better.

Reply:
Also, I'm not sure, can has_perm accept a list? Like `checker.has_perm(Team.PERMISSION_VIEW, team)`. If it can then we should definately not loop.  You've probably already checked this, but figured I would mention it just in case.